BlinkID/example/lib/main.dart

```diff
@@ -0,0 +1,103 @@
+import 'package:blinkid_flutter/microblink_scanner.dart';
+import 'package:flutter/material.dart';
+import 'package:sample/document_scanner.dart';
+
+void main() {
+  runApp(const MyApp());
+}
+
+class MyApp extends StatelessWidget {
+  const MyApp({super.key});
+
+  // This widget is the root of your application.
+  @override
+  Widget build(BuildContext context) {
+    return MaterialApp(
+      title: 'Flutter Demo',
+      theme: ThemeData(
+        colorScheme: ColorScheme.fromSeed(seedColor: Colors.deepPurple),
+        useMaterial3: true,
+      ),
+      home: const HomePage(),
+    );
+  }
+}
+
+class HomePage extends StatefulWidget {
+  const HomePage({super.key});
+
+  @override
+  State<HomePage> createState() => _HomePageState();
+}
+
+class _HomePageState extends State<HomePage> {
+  @override
+  Widget build(BuildContext context) {
+    return Scaffold(
+      appBar: AppBar(title: const Text('BlinkID Example')),
+      body: Center(
+        child: FilledButton(
+          onPressed: () async {
+            final result = await context.scanDocument();
+            if (!context.mounted || result == null) return;
+
+            showDialog(
```

Comment:
When moving `onPressed` callback to a separate function, showing of the dialog should also be moved to a separate function.

```diff
diff --git a/BlinkID/example/lib/main.dart b/BlinkID/example/lib/main.dart
index 3951206..b106b20 100644
--- a/BlinkID/example/lib/main.dart
+++ b/BlinkID/example/lib/main.dart
@@ -31,42 +31,46 @@ class HomePage extends StatefulWidget {
 }
 
 class _HomePageState extends State<HomePage> {
+  void _onScanDocumentButtonPressed() {
+    context.scanDocument().then((result) => context.mounted ? _handleScanResult(result) : null);
+  }
+
+  void _handleScanResult(List<RecognizerResult>? result) => result == null
+      ? null
+      : showDialog(
+          context: context,
+          builder: (_) => Dialog(
+            clipBehavior: Clip.antiAlias,
+            child: Scaffold(
+              appBar: AppBar(title: const Text('Result'), leading: const CloseButton()),
+              body: ListView.builder(
+                itemBuilder: (_, index) {
+                  final item = result[index];
+
+                  final name = switch (item) {
+                    BlinkIdMultiSideRecognizerResult _ => 'BlinkIdMultiSideRecognizerResult',
+                    _ => item.runtimeType.toString(),
+                  };
+
+                  return ExpansionTile(
+                      title: Text(name),
+                      children: item.formatted.entries.map((entry) {
+                        return ListTile(title: Text('${entry.key}: ${entry.value}'));
+                      }).toList());
+                },
+                itemCount: result.length,
+              ),
+            ),
+          ),
+        );
+
   @override
   Widget build(BuildContext context) {
     return Scaffold(
       appBar: AppBar(title: const Text('BlinkID Example')),
       body: Center(
         child: FilledButton(
-          onPressed: () async {
-            final result = await context.scanDocument();
-            if (!context.mounted || result == null) return;
-
-            showDialog(
-                context: context,
-                builder: (_) => Dialog(
-                      clipBehavior: Clip.antiAlias,
-                      child: Scaffold(
-                        appBar: AppBar(title: const Text('Result'), leading: const CloseButton()),
-                        body: ListView.builder(
-                          itemBuilder: (_, index) {
-                            final item = result[index];
-
-                            final name = switch (item) {
-                              BlinkIdMultiSideRecognizerResult _ => 'BlinkIdMultiSideRecognizerResult',
-                              _ => item.runtimeType.toString(),
-                            };
-
-                            return ExpansionTile(
-                                title: Text(name),
-                                children: item.formatted.entries.map((entry) {
-                                  return ListTile(title: Text('${entry.key}: ${entry.value}'));
-                                }).toList());
-                          },
-                          itemCount: result.length,
-                        ),
-                      ),
-                    ));
-          },
+          onPressed: _onScanDocumentButtonPressed,
           child: const Text('Scan Document'),
         ),
       ),
```